packages/starknet-snap/src/index.ts

```diff
@@ -293,19 +295,18 @@ export const onRpcRequest: OnRpcRequestHandler = async ({ request }) => {
         return await getStarkName(apiParams);
 
       default:
-        throw new Error('Method not found.');
+        throw new MethodNotFoundError() as unknown as Error;
     }
   } catch (error) {
-    // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
-    logger.error(`Error: ${error}`);
-    // We don't want to expose the error message to the user when it is a production build
-    if (logger.logLevel === LogLevel.OFF) {
-      throw new SnapError(
-        'Unable to execute the rpc request',
-      ) as unknown as Error;
-    } else {
-      throw new SnapError(error.message) as unknown as Error;
+    let snapError = error;
+
+    if (!isSnapRpcError(error)) {
+      snapError = new SnapError('Unable to execute the rpc request');
     }
+    logger.error(
+      `onRpcRequest error: ${JSON.stringify(snapError.toJSON(), null, 2)}`,
```

Comment:
Shall we add a test for this logger.error to ensure that it's shown when log level is error ?

Reply:
no need, as the logger test is already tested the feature